### ISSUE TITLE
Pretrained window/hybrid SABlock backbone model for Detection task

### ISSUE THREAD

**Issue by CanyonWind** [Sense-X/UniFormer#12]
Hi, thank you for the contribution to this super-rad work!

Wonder that, in your experiments, whether the backbone models used in Detection task with stage-3 window/hybrid SABlock (S-h14, B-h14) are needed to be pretrained on imagenet? 

If so, could these backbones with window/hybrid SABlock be released? And if not, are the weights loaded directly from the regular model with global window in stage-3?

Thanks!

**Comment by Andy1621** [comment]
Sorry for the late response.
Actually, the backbones used in downstream tasks with window/hybrid SABlock are **not need** to be pre-tained on ImageNet. The weights are directly loaded.

Since we train the backbone with **224x224** images on ImageNet, the normal self-attention in stage-3 **can be seen as window self-attention with size 14x14** (224/16=14). 
Hence, for the downstream tasks, we only need to ensure the window in stage-3 is larger than 14x14, thus the pre-trained weights can be well utilized.

**Comment by Andy1621** [comment]
We do not use backbones trained with Token Labeling for **fair comparisons**, since most SOTA models are trained without it.

As for layer scale, it is good for **avoiding NaN for ImageNet**. We have tried to used backbones with Layer Scale for video classification, but **it sometimes can not converge in the later epochs**.  It seems strange but it does not affect other downstream tasks. 

Actually, most of the current works used Layer Scale for training large models. I think it is a better choice to use it. 
Moreover,  backbones trained with Token Labeling also help downstream tasks like detection. If you only pursue better results, you can use those powerful models with Token Labeling.

**Comment by CanyonWind** [comment]
And when using the backbone in the detection task, in `UniFormer` module, each stage initializes a new `norm1-4` layer and discards the original `self.norm` layer (the one before the final classification head). Could you please share some insights on this implementation as well? Thanks a lot!

**Comment by Andy1621** [comment]
@CanyonWind Thanks for your detailed questions!

About the two techniques:
- Yes, the model trained with token labeling will only have one difference than the regular model, since it need to generate dense labels for all tokens. You can find it in the [original repo](https://github.com/zihangJiang/TokenLabeling/blob/f291cbffdc48e37c248fc4a9ab1b7884c85c997f/tlt/models/lvvit.py#L131-L138). Actually for downstream tasks, the classification head is not used anymore. Researchers only utilized the intermediate features to form FPN. Thus for backbone trained with Token Labeling, **the additional auxiliary head and the classification head are both removed**.
- **For smaller models, it doesn't help for accuracy improvement but it helps for normal convergence**. Note that all vision transformers are easier to be NaN when training. The layer Scale is a smart technique to avoid this problem. For larger models, Layer Scale is often used for better convergence. You can find the results in the [original paper (CaiT)](https://arxiv.org/abs/2103.17239).

About `norm` for detection task:
- Actually, the extra `norm` layers are popular settings for detection tasks. We simply follow MMDetection to add them. You can find them in [ResNet](https://github.com/open-mmlab/mmdetection/blob/7a9bc498d5cc972171ec4f7332afcd70bb50e60e/mmdet/models/backbones/resnet.py#L567-L597) and [Swin](https://github.com/open-mmlab/mmdetection/blob/7a9bc498d5cc972171ec4f7332afcd70bb50e60e/mmdet/models/backbones/swin.py#L638-L641).

**Comment by CanyonWind** [comment]
Thanks for the clarifications. These are awesome!

**Comment by Andy1621** [comment]
@CanyonWind We have verified that Token Labeling can help detection. Have a try!
